### ISSUE TITLE
Blueprint Block

### ISSUE THREAD

**Issue by RogueRedHawk** [Anuken/Mindustry-Suggestions#93]
**Describe the content.** Blueprint is a block that holds a schematic. It is a 1x1 block and when supplied with power, it projects a schematic into "ghost form" like what phantom builder drones can build.

**How does the blueprint know where to "project" the schematic?**  When a Blueprint is placed down, it can be set to anchor mode. When a schematic is created, a check is made to ensure only ONE Blueprint set to "anchor" is being saved. When a Blueprint is placed down, only schematics with an anchor are able to be placed. That "anchor" in the schematic is then placed where the Blueprint is when called on. Notice this mechanic allows nested Blueprints.

**How do you think this content will improve the game?** This mechanic will introduce a level of automation to the game that currently doesn't exist. Modular bases, self expanding bases, bases that gradually become more complex. Additionally, Blueprints would give another great use to Phantom Builder drones, as you could place down a nested schematic design for a power plant and you Phantom Builder could work on power while you build on defense. The Blueprint could also work well when logic blocks come out in the future, depending on how Anuke decides to do them.

TL;DR: This Blueprint mechanic would be healthy for the game in my opinion because it is optional (you don't have to use them if you don't want to), minimal (only one block added), relatively simple (simple mechanics, not a lot of rules to remember), and dynamic (bang for buck).

1. - [x] I have done a quick search in the list of suggestions to make sure this has not been suggested yet. -- Searched for "Blueprint".

2. - [x] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet. -- I've done a bunch of modding so I've read through every item in the game and properties multiple times..

**Comment by Anuken** [comment]
I don't think this needs to be a separate block. The logic system will be able to handle queuing of blocks.
